server/aws-lsp-codewhisperer/src/language-server/netTransform/transformHandler.ts

```diff
@@ -0,0 +1,255 @@
+import path = require('path')
+import { Workspace } from '@aws/language-server-runtimes/out/features'
+import * as fs from 'fs'
+import * as os from 'os'
+import { v4 as uuidv4 } from 'uuid'
+import {
+    CreateUploadUrlResponse,
+    GetTransformationRequest,
+    StopTransformationRequest,
+} from '../../client/token/codewhispererbearertokenclient'
+import { CodeWhispererServiceToken } from '../codeWhispererService'
+import { getCWStartTransformRequest, getCWStartTransformResponse } from './converter'
+import {
+    CancellationJobStatus,
+    QNetCancelTransformRequest,
+    QNetCancelTransformResponse,
+    QNetGetTransformPlanRequest,
+    QNetGetTransformPlanResponse,
+    QNetGetTransformRequest,
+    QNetGetTransformResponse,
+    QNetStartTransformRequest,
+    QNetStartTransformResponse,
+} from './models'
+import { cleanup, createZip, getSha256 } from './utils'
+import fetch = require('node-fetch')
+
+export class TransformHandler {
+    private client: CodeWhispererServiceToken
+    private workspace: Workspace
+    constructor(client: CodeWhispererServiceToken, workspace: Workspace) {
+        this.client = client
+        this.workspace = workspace
+    }
+
+    async startTransformation(userInputrequest: QNetStartTransformRequest): Promise<QNetStartTransformResponse> {
+        try {
+            const uploadId = await this.preTransformationUploadCode(userInputrequest)
+            const request = getCWStartTransformRequest(userInputrequest, uploadId)
+            console.log('send request to start transform api: ' + JSON.stringify(request))
+            const response = await this.client.codeModernizerStartCodeTransformation(request)
+            console.log('response start transform api: ' + JSON.stringify(response))
+            return getCWStartTransformResponse(response, uploadId)
+        } catch (error) {
+            const errorMessage = (error as Error).message ?? 'Error in StartTransformation API call'
+            console.log(errorMessage)
+            throw new Error(errorMessage)
+        }
+    }
+
+    async preTransformationUploadCode(userInputrequest: QNetStartTransformRequest): Promise<string> {
+        let uploadId = ''
+        const randomPath = uuidv4().substring(0, 4)
+        const workspacePath = path.join(os.tmpdir(), randomPath)
```

Comment:
There is a `workspace.fs.getTempDirPath` method that does this in a cross-platform way: 

https://github.com/aws/language-server-runtimes/blob/main/src/features/workspace.ts#L25C5-L25C19